docs/dev/how-to/estimate-gas.md

```diff
@@ -0,0 +1,399 @@
+
+const zkSyncProvider = new Provider("https://testnet.era.zksync.dev");
+const ethereumProvider = ethers.getDefaultProvider("goerli");
+const wallet = new Wallet(TEST_PRIVATE_KEY, zkSyncProvider, ethereumProvider);
+
+const gasPrice = await wallet.providerL1!.getGasPrice();
+
+// The calldata can be encoded in the same way as for Ethereum
+const calldata = "0x...";
+const gasLimit = BigNumber.from(1000);
+const gasPerPubdataByte = BigNumber.from(800);
+
+const txCostPrice = await wallet.getBaseCost({
+  gasPrice,
+  gasLimit,
+  gasPerPubdataByte
+});
+
+console.log(`Executing the transaction will cost ${ethers.utils.formatEther(txCostPrice)} ETH`);
+
+// Initiate L2 transfer via L1 and execute from zkSync Era wallet
+const someL2Contract = "0x19a5bfcbe15f98aa073b9f81b58466521479df8d";
+const executeTx = await wallet.requestExecute({
+  calldata,
+  l2GasLimit: gasLimit,
+  gasPerPubdataByte,
+  contractAddress: someL2Contract,
+  overrides: {
+    gasPrice,
+    value: txCostPrice,
+  },
+});
+
+await executeTx.wait();
+```
+@tab Python
+```sh
+// In progress. Check back later.
+```
+@tab Go
+```sh
+// In progress. Check back later.
+```
+@tab Java
+```sh
+// In progress. Check back later.
+```
+:::
+
+### Result
+
+6. The return value is a 256-bit unsigned integer in hexadecimal representing how much gas the transaction uses.
+
+```json
+{
+    "jsonrpc": "2.0",
+    "result": "0x25f64db",
+    "id": 2
+}
+```
+
+
+
+## L2 transactions
+
+Gas estimation for L2 to L2 transactions on zkSyncEra works in the same way as in Ethereum.
+
+Supply the same struct as for a L1 to L2 transaction.
+
+?? // can you add some code here please
+
+
+
+## L2 to L1 transactions
+
+Two transactions are required: 
+
+- An L2 transaction which sends a message of arbitrary length. 
+- When the block the message resides in is proved on L1, verification and confirmation takes place.
+
+For example, an L2 transaction uses a bridge to transfer funds. The tokens are removed from L2, and a message is sent to L1 saying the tokens were burned. On L1, the block containing the message is proved, and the bridge counterpart verifies that the token was indeed present in a certain block on L2 and supplies the funds.
+
+1. Import the zkSync Era library or contract containing the required functionality. 
+
+2. Get a `Contract` object that represents the [`Messenger`](../developer-guides/system-contracts.md#l1messenger) contract.
+
+3. Transform the request into a raw bytes array.
+
+4. Use the [`sendToL1`](https://github.com/matter-labs/v2-testnet-contracts/blob/b8449bf9c819098cc8bfee0549ff5094456be51d/l2/system-contracts/interfaces/IL1Messenger.sol#L5) function from the `IL1Messenger.sol` interface, passing the message as a raw bytes array.
+
+```solidity
+function sendToL1(bytes memory _message) external returns (bytes32);
+```
+
+:::warning What is a message?
+- A message functions like an event on Ethereum. The difference is that a message publishes data on L1. 
```

Comment:
```suggestion
- A message functions similar to event on Ethereum. The difference is that a message publishes data on L1. 
```

I think they are not like each other but rather have some similarities

Reply:
"like" is equivalent here, if you use "similar to" you have to say .... "A message functions in a similar way to an event" and that's a lot of extra words :)